### PR TITLE
Add domOrder direction in FocusZone

### PR DESCRIPTION
**Pull request by sophieH29**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

### TL;DR
Usually, when screen reader users navigate to menus or lists, they want to use Up/Down arrow keys as the semantic of these components stands for. In real-world, we have a lot of examples, where we have a horizontal list/menu or a grid with items. Keyboard users (not visually impaired people) intentionally will want to use Left/Right arrow keys to navigate within these components.

To achieve consistency, Right/Down arrow keys should navigate to the next focusable item in the component. Left/Up arrow keys - to the previous focusable item. Note: in RTL mode, only left and arrow keys are switched.

#### Why not just use bidirectional direction?

1. For some Grid-like components (for example, emoji picker), screen readers are not aware of the grid so they would use Up/Down arrow keys to navigate through emoji as in the list. Keyboard users most probably will use Left/Right arrow keys to navigate in the grid, the focus will move horizontally row by row.
2. It has less impact on performance as it doesn't calculate elements' client distance to calculate where to move focus next

### Changes
- Add one more direction to `FocusZone`'s enum - `domOrder`
- Add example of a horizontal menu with `domOrder` direction

**Note**: It might be also useful to implicitly set `domOrder` direction for some components in the library.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10324)

### CODE REVIEW

**Comment by dzearing** [comment]
I see all the changes I'd expect... except where domOrder is used. Is there a missing FocusZone.tsx change in this PR or am I just missing it?

**Comment by kenotron** [comment]
@JasonGore - looks like you're owner here. I'm adding automerge here to assist merging
